### ISSUE TITLE
Problem with upgrading to 23.4.0

### ISSUE THREAD

**Issue by afaianswq** [getsentry/self-hosted#2127]
### Self-Hosted Version

23.4.0

### CPU Architecture

x86_64

### Docker Version

23.0.3

### Docker Compose Version

2.17.2

### Steps to Reproduce

1. Make sure your postgres data directory size is larger that free disk space
2. Upgrade from 23.3.1 to 23.4.0

### Expected Result

Receive an error explaining that during the upgrade a postgres data directory will be **copied** and there is not enough free disk space for this operation.

### Actual Result

Upgrade fails when system runs out of free disk space:
```
error while copying relation "pg_toast.pg_toast_19429": could not write file "/var/lib/postgresql/14/data/base/16401/19432.188": No space left on device
Failure, exiting
Error in install/upgrade-postgres.sh:6.
'docker run --rm -v sentry-postgres:/var/lib/postgresql/9.6/data -v sentry-postgres-new:/var/lib/postgresql/14/data tianon/postgres-upgrade:9.6-to-14' exited with status 1
-> ./install.sh:main:33
--> install/upgrade-postgres.sh:source:6
```

**Comment by hubertdeng123** [comment]
It should be possible to upgrade without copying the files with the `--link` option that is available. You can tweak the upgrade script to use that. This is not enabled by default, but should be safe to use

**Comment by afaianswq** [comment]
> It should be possible to upgrade without copying the files with the `--link` option that is available. You can tweak the upgrade script to use that. This is not enabled by default, but should be safe to use

Thanks for the hint, however this is a bug report, and the fact that install script doesn't make sure that there is enough free disk space before starting upgrade, which involves copying postgres database, is clearly a bug. In my case, upgrade has already failed and I had a very hard time recovering from this failure.

**Comment by afaianswq** [comment]
Also, volume copying seems to be hardcoded here: https://github.com/getsentry/self-hosted/blob/d8b143286398d51bb14da188381ae229d8c1643b/install/upgrade-postgres.sh#L17

**Comment by hubertdeng123** [comment]
Sorry to hear you've had trouble with the upgrade. We always try to be thoughtful when it comes to making changes, and in the [23.4.0 release notes](https://github.com/getsentry/sentry/releases/tag/23.4.0) we did include a note that the upgrade may require additional disk space. We've provided a script that hopefully works for most people, and if you'd like to suggest some changes to it you're always free to open a PR.
